host/Manager.ts

```diff
@@ -6,17 +6,89 @@ import {
   WebSocketServer
 } from '@stencila/executa'
 import { getLogger } from '@stencila/logga'
-import { isA, Node, SoftwareSession } from '@stencila/schema'
+import {
+  Environment,
+  isA,
+  Mount,
+  Node,
+  SoftwareSession,
+  mount
+} from '@stencila/schema'
 import crypto from 'crypto'
 import { DockerSession } from './DockerSession'
 import { FirecrackerSession } from './FirecrackerSession'
 import { Session } from './Session'
 
 const log = getLogger('sparkla:manager')
+
 export interface SessionType {
   new (): FirecrackerSession | DockerSession
 }
 
+/**
+ * The properties defined by the JWT.
+ */
+interface JwtLimits {
+  mem?: number
+  cpu?: number
+  vol?: [string, string, string][]
+}
+
+/**
+ * A slightly different interface than SoftwareSession
+ */
+export interface SparklaSoftwareSession {
+  environment: Environment
+  cpuRequested?: number
+  cpuLimit?: number
+  memoryRequested?: number
+  memoryLimit?: number
+  volumeMounts?: Mount[]
+}
+
+function optionalMin(a?: number, b?: number): number | undefined {
+  if (a === undefined && b === undefined) return undefined
+  if (a === undefined) return b
+  if (b === undefined) return a
+
+  return Math.min(a, b)
+}
+
+function applySessionLimits(
+  session: SoftwareSession,
+  limits: JwtLimits
```

Comment:
As discussed, let's get rid of `JwtLimits` and just make this `limits` a `SoftwareSession` with `memoryLimit` set